snt/hbhi/set_up_general.py

```diff
@@ -436,10 +446,10 @@ def set_habitats(config, manifest, hdf, lhdf, archetype_ds, hab_multiplier, my_d
                                                                                      334],
                                                                            "Values": my_spline})
         malaria_config.set_species_param(config, sp, "Habitats", linear_spline_habitat, overwrite=True)
```

Comment:
Current code would cause EMOD to complain that it is expecting an array for `Habitats`. I took a look at the config.json and the offending bit is:
```
        "Habitats": {
          "Habitat_Type": "CONSTANT",
          "Max_Larval_Capacity": 206516.3487606087
        },
```
So two things, Habitats need to be a list, and current code overwrites the spline setting by constant. (This should be a list of two, one with `LINEAR_SPLINE` and one with `CONSTANT`). I made some modification to this part and the simulations can now run with two habitat types :
```
    for (s, sp) in zip(fraction, ['arabiensis', 'funestus', 'gambiae']):
        linear_spline_habitat = malaria_config.configure_linear_spline(manifest,
                                                                       max_larval_capacity=pow(10,
                                                                                               maxvalue) * s * hab_multiplier,
                                                                       capacity_distribution_number_of_years=1,
                                                                       capacity_distribution_over_time={
                                                                           "Times": [0, 31, 59, 90, 120, 151, 181, 212,
                                                                                     243, 273, 304,
                                                                                     334],
                                                                           "Values": my_spline})
        const_habitat = dfs.schema_to_config_subnode(manifest.schema_file, ["idmTypes", "idmType:VectorHabitat"])
        const_habitat.parameters.Habitat_Type = "CONSTANT"
        const_habitat.parameters.Max_Larval_Capacity = pow(10, const) * s * const_mult
        new_habitat = [linear_spline_habitat, const_habitat.parameters]
        malaria_config.set_species_param(config, sp, "Habitats", new_habitat, overwrite=True)
```